pkg/models/repository.go

```diff
@@ -0,0 +1,45 @@
+package models
+
+import (
+	"time"
+
+	"gorm.io/gorm"
+)
+
+// TODO Review the content for this table.
+type Repository struct {
+	Base
+	// Repository URL
+	URL string `json:"url" gorm:"not null"`
+	// Last time the repo meta data was read
+	LastReadTime *time.Time `gorm:"default:null"`
+	// Last time the repo meta data failed to be read
+	LastReadError *string `gorm:"default:null"`
+	// ReferRepoConfig to Repository UUID
+	ReferRepoConfig *string `gorm:"default:null"`
+	// RepoConfig is the repository configuration
+	RepoConfig *RepositoryConfiguration `gorm:"foreignKey:UUID;references:ReferRepoConfig"`
```

Comment:
which relationship is this creating with RepositoryConfiguration, "has one" or "belongs to"?

Reply:
Thanks mate! Updated! Initially I understood the relationships of the entities differently; Actually, one RepositoryConfiguration "has one" Repository; one Repository "has many" Repository; I have updated the name of the fields that reference the foreign keys to match the gorm conventions.

Reply:
several updates here; I have related all the models and from the related one, can be loaded the slice or related record by using `Preload` method from gorm.
- `RepositoryConfig` belongs to `Repository`.
- `Repository` has many `RepositoryConfig`.
- `Repository` has many `Rpms`.
- `Rpms` has many `Repository`.